packages/ui/src/ui/store/actions/system/masters.ts

```diff
@@ -399,6 +403,96 @@ export function loadMasters() {
     };
 }
 
+const getPathByMasterType = (type: string): string | null => {
+    switch (type) {
+        case 'primary':
+        case 'secondary':
+            return '//sys/cluster_masters';
+        case 'providers':
+            return '//sys/timestamp_providers';
+        case 'queue_agent':
+            return '//sys/queue_agents/instances';
+        default:
+            return null;
+    }
+};
+
+export const changeMaintenance = ({
+    path,
+    login,
+    maintenance,
+    message,
+}: {
+    path: string;
+    login: string;
+    maintenance: boolean;
+    message: string;
+}) =>
+    ytApiV3Id.executeBatch(YTApiId.systemMastersMaintenance, {
+        requests: [
+            {
+                command: 'set',
+                parameters: {
+                    path: `${path}/@maintenance`,
+                },
+                input: maintenance,
+            },
+            {
+                command: 'set',
+                parameters: {
+                    path: `${path}/@maintenance_message`,
+                },
+                input: maintenance
+                    ? `Maintenance was set by ${login} at ${new Date().toISOString()} from UI${message ? `. ${message}` : ''}`
+                    : '',
+            },
+        ],
+    });
+
+export const changeMasterMaintenance =
+    ({
+        address,
+        maintenance,
+        message,
+    }: {
+        address: string;
+        maintenance: boolean;
+        message: string;
+    }): ThunkAction<any, RootState, unknown, MastersStateAction> =>
+    async (dispatch, getSate) => {
+        const state = getSate();
+        const {primary, secondary, queueAgents, providers} = state.system.masters;
+        const instances: MasterInstance[] = [
+            ...primary.instances,
+            ...secondary.reduce<MasterInstance[]>((acc, item) => [...acc, ...item.instances], []),
+            ...queueAgents.instances,
+            ...providers.instances,
+        ];
+
+        const master = instances.find((i) => {
+            const {host, physicalHost} = i.toObject();
+            return host === address || physicalHost === address;
```

Comment:
It is more strict to check only corresponding value according to current mode (`host`/`container`). Also it explicitly highlights dependencies in the code.